subsys/bluetooth/controller/Kconfig

```diff
@@ -157,7 +157,7 @@ config BT_CTLR_DUP_FILTER_LEN
 
 config BT_CTLR_DUP_FILTER_ADV_SET_MAX
 	int "Number of Extended Advertising Sets in the scan duplicate filter"
-	depends on BT_OBSERVER && (BT_CTLR_DUP_FILTER_LEN > 0)
+	depends on BT_OBSERVER && BT_CTLR_ADV_EXT && (BT_CTLR_DUP_FILTER_LEN > 0)
```

Comment:
Why does this need to depend on `BT_CTLR_ADV_EXT`? Are we not using the same duplicate filter list for legacy scanning?

Reply:
This value is used only when Extended Scanning is supported. Additional values like DID, data status etc. is maintained to filter duplicate Extended Advertising PDUs from same peer device that uses multiple advertising sets.